packages/fast-components/src/switch/switch.ts

```diff
@@ -0,0 +1,100 @@
+import { attr, observable } from "@microsoft/fast-element";
+import { FormAssociated } from "../form-associated";
+import { keyCodeSpace } from "@microsoft/fast-web-utilities";
+import { bool } from "../utilities";
+
+export class Switch extends FormAssociated<HTMLInputElement> {
+    @attr({ attribute: "readonly" })
+    public readOnly: boolean; // Map to proxy element
+    private readOnlyChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.readOnly = this.readOnly;
+        }
+
+        this.readOnly
+            ? this.classList.add("readonly")
+            : this.classList.remove("readonly");
+    }
+
+    @attr({ attribute: "checked" })
+    public checkedAttribute: string | null;
+    private checkedAttributeChanged(): void {
+        this.defaultChecked = typeof this.checkedAttribute === "string";
+    }
+
+    /**
+     * Initialized to the value of the checked attribute. Can be changed independently of the "checked" attribute,
+     * but changing the "checked" attribute always additionally sets this value.
+     */
+    @observable
+    public defaultChecked: boolean = !!this.checkedAttribute;
+    private defaultCheckedChanged(): void {
+        if (!this.dirtyChecked) {
+            // Setting this.checked will cause us to enter a dirty state,
+            // but if we are clean when defaultChecked is changed, we want to stay
+            // in a clean state, so reset this.dirtyChecked
+            this.checked = this.defaultChecked;
+            this.dirtyChecked = false;
+        }
+    }
+
+    /**
+     * The checked state of the control
+     */
+    @observable
+    public checked: boolean = this.defaultChecked;
+    private checkedChanged(): void {
+        if (!this.dirtyChecked) {
+            this.dirtyChecked = true;
+        }
+
+        this.updateForm();
+
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.checked = this.checked;
+        }
+
+        if (this.constructed) {
+            this.dispatchEvent(
+                new CustomEvent("change", { bubbles: true, composed: true })
+            );
+        }
+
+        this.checked ? this.classList.add("checked") : this.classList.remove("checked");
+    }
+
+    protected proxy = document.createElement("input");
```

Comment:
You'll want to make sure this input has it's `type` attribute set to `"checkbox"`